src/apps/jsgeneric/jsgeneric.cpp

```diff
@@ -64,24 +73,45 @@ namespace ccfapp
           args);
         */
 
-        const auto err_it = response.find(jsonrpc::ERR);
+        auto err_it = response.find("error");
         if (err_it == response.end())
         {
-          const auto result_it = response.find(jsonrpc::RESULT);
+          auto result_it = response.find("result");
           if (result_it == response.end())
           {
-            // Response contains neither RESULT nor ERR. It may not even be an
+            // Response contains neither result nor error. It may not even be an
             // object. We assume the entire response is a successful result.
-            return make_pair(true, response);
+            args.rpc_ctx->set_response_result(std::move(response));
+            return;
           }
           else
           {
-            return make_pair(true, *result_it);
+            args.rpc_ctx->set_response_result(std::move(*result_it));
+            return;
           }
         }
         else
         {
-          return make_pair(false, *err_it);
+          int err_code = jsonrpc::CCFErrorCodes::SCRIPT_ERROR;
+          std::string msg = "";
+
+          if (err_it->is_object())
+          {
+            auto err_code_it = err_it->find("code");
```

Comment:
I think they'll go away in future, once we have #705. So instead of this `"code"` needing to match the name of a field in a helper in the `__environment` script, as we currently do, we'll just be able to call `ctx.set_error(code, msg)` (or similar) directly from Lua/JS.